src/components/query-viewer-pivot-table/query-viewer-pivot.tsx

```diff
@@ -98,7 +98,10 @@ export class QueryViewerPivot {
     if (!oldValue) {
       return;
     }
-    setPageDataForTable(this.queryViewerConfiguration.oat_element, newValue);
+    if (newValue !== oldValue) {
+      setPageDataForTable(this.queryViewerConfiguration.oat_element, newValue);
+      this.renderPivot();
+    }
```

Comment:
```suggestion
    // This is a WA since the Table does not refresh its pageSize if we don't update
    // the pageData previously
    setPageDataForTable(this.queryViewerConfiguration.oat_element, newValue);
    this.renderPivot();
```